Player/Player.gd

```diff
@@ -0,0 +1,33 @@
+extends CharacterBody2D
+
+
+const SPEED = 200
+@onready var anim = get_node("AnimatedSprite2D")
+
+func _ready():
+	anim.play("Idle")
+
+func _physics_process(delta):
+	var velocity = Vector2()
+	
+	if Input.is_action_pressed("right"):
+		velocity.x += 1
+		anim.flip_h = false
+		
+	if Input.is_action_pressed("left"):
+		velocity.x -= 1
+		anim.flip_h = true
+		
+	if Input.is_action_pressed("down"):
+		velocity.y += 1
+		
+	if Input.is_action_pressed("up"):
+		velocity.y -= 1
+		
+	if velocity.length() > 0:
+		velocity = velocity.normalized() * SPEED
+		anim.play("Running")
+	else:
+		anim.play("Idle")
+	position += velocity * delta
```

Comment:
Input detection should primarily be done in `_process`, which is called every frame. The `_physics_process` function should focus on physics-related logic, such as character movement and physics-based simulations.

It's best practice to detect input in `_process` and then update character movement or other actions in `_physics_process`.

This keeps the input detection and physics-related logic separated, ensuring that character movement aligns with the game's physics.


You can create a "global" variable to detect the player `velocity`. (Just a note, `CharacterBody2D`, already has [`velocity` property](https://docs.godotengine.org/en/stable/classes/class_characterbody2d.html#class-characterbody2d-property-velocity))

Some help:

```suggestion
var input_direction = Vector2.Zero # Variable name example

func _process(_delta):
	if Input.is_action_pressed("right"):
		input_direction.x += 1
		anim.flip_h = false
	if Input.is_action_pressed("left"):
		input_direction.x -= 1
		anim.flip_h = true
	if Input.is_action_pressed("down"):
		input_direction.y += 1
	if Input.is_action_pressed("up"):
		input_direction.y -= 1
		
	if input_direction.length() > 0:
		input_direction = input_direction.normalized() * SPEED
		anim.play("Running")
	else:
		anim.play("Idle")

func _physics_process(delta):
	position += input_direction * delta
	# Or move_and_slide(input_direction * SPEED)
```

Not tested, but it should be something like this 😁